llmfoundry/callbacks/generate_callback.py

```diff
@@ -74,9 +75,11 @@ def generate(self, state: State, logger: Logger):
         dummy_input = device.tensor_to_device(dummy_input)
         with get_precision_context(state.precision):
             with torch.no_grad():
+                assert isinstance(model.model, torch.nn.Module)
                 _ = model.model(input_ids=dummy_input)
 
-            output_token_ids = model.model.generate(
+            assert hasattr(model.model, 'generate')
+            output_token_ids = model.model.generate(  # type: ignore
```

Comment:
If you checked the attribute in the previous line, why should this be a type ignore?

Reply:
That is a great question that I could not figure out 😅 it was only python 3.9 that it failed, not 3.10. so I conceded on this one.